pyfr/backends/base/types.py

```diff
@@ -170,7 +205,13 @@ def offset(self):
         if 'bank' in self.tags:
             raise AttributeError('offset undefined for banked slices')
 
-        return self.parent.offset + self.ra*self.pitch + self.ca*self.itemsize
+        if self.backend.blocks:
+            _offset = self.itemsize*(self.ra*self.leaddim
+                                     + self.ba*self.blocksz)
+        else:
+            _offset = self.ra*self.pitch + self.ca*self.itemsize
+
+        return self.parent.offset + _offset
```

Comment:
Might be able to clean up by taking a factor of self.itemsize out of _offset (note the definition of pitch).